src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -9812,6 +9905,312 @@ public UnmanagedCallersOnlyAttribute()
+                    Diagnostic(ErrorCode.ERR_WrongFuncPtrCallingConvention, "M", isSuppressed: false).WithArguments("C.M()", "FastCall").WithLocation(15, 60)
+                    );
+            }
+
+            if (diagnosticToSkip != 6)
+            {
+                diagnostics.Add(
+                    // (16,72): error CS8786: Calling convention of 'C.M()' is not compatible with 'Unmanaged'.
+                    //         delegate* unmanaged[Cdecl, Thiscall]<void> ptrCdeclThiscall = &M;
+                    Diagnostic(ErrorCode.ERR_WrongFuncPtrCallingConvention, "M", isSuppressed: false).WithArguments("C.M()", "Unmanaged").WithLocation(16, 72)
+                    );
+            }
+
+
+            comp.Assembly.SetOverrideRuntimeSupportsUnmanagedSignatureCallingConvention();
+            comp.VerifyDiagnostics(diagnostics.ToArray());
+        }
+
+        [Fact]
+        public void UnmanagedCallersOnlyAttribute_AddressOfUsedInAttributeArgument()
+        {
+            var comp = CreateCompilationWithFunctionPointers(new[] { @"
+using System;
+using System.Runtime.InteropServices;
+[AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+unsafe class Attr : Attribute
+{
+    public Attr() {}
+
+    public delegate* unmanaged<void> PropUnmanaged { get; set; }
+    public delegate*<void> PropManaged { get; set; }
+    public delegate* unmanaged[Cdecl]<void> PropCdecl { get; set; }
+}
+unsafe class C
+{
+    [UnmanagedCallersOnly]
+    static void M1()
+    {
+    }
+
+    [Attr(PropUnmanaged = &M1)]
+    [Attr(PropManaged = &M1)]
+    [Attr(PropCdecl = &M1)]
+    static unsafe void M2()
+    {
+
+    }
+}
+", UnmanagedCallersOnlyAttribute });
+
+            comp.Assembly.SetOverrideRuntimeSupportsUnmanagedSignatureCallingConvention();
+            comp.VerifyDiagnostics(
+                // (20,11): error CS0655: 'PropUnmanaged' is not a valid named attribute argument because it is not a valid attribute parameter type
+                //     [Attr(PropUnmanaged = &M1)]
+                Diagnostic(ErrorCode.ERR_BadNamedAttributeArgumentType, "PropUnmanaged", isSuppressed: false).WithArguments("PropUnmanaged").WithLocation(20, 11),
+                // (21,11): error CS0655: 'PropManaged' is not a valid named attribute argument because it is not a valid attribute parameter type
+                //     [Attr(PropManaged = &M1)]
+                Diagnostic(ErrorCode.ERR_BadNamedAttributeArgumentType, "PropManaged", isSuppressed: false).WithArguments("PropManaged").WithLocation(21, 11),
+                // (22,11): error CS0655: 'PropCdecl' is not a valid named attribute argument because it is not a valid attribute parameter type
+                //     [Attr(PropCdecl = &M1)]
+                Diagnostic(ErrorCode.ERR_BadNamedAttributeArgumentType, "PropCdecl", isSuppressed: false).WithArguments("PropCdecl").WithLocation(22, 11)
+            );
+        }
```

Comment:
>} [](start = 8, length = 1)

Consider testing:
```C#
using System;
using System.Runtime.InteropServices;
[AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
unsafe class Attr : Attribute
{
    public Attr(delegate*<void> d) {}
}
unsafe class C
{
    [UnmanagedCallersOnly]
    [Attr(&M1)]
    static void M1()
    {
    }
}
```

Reply:
Consider testing:
```C#
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
public unsafe class C
{
    [UnmanagedCallersOnly(CallConvs = new[] { typeof(CallConvFastcall) })]
    static void F(int i = G(&F)) { }
    static int G(delegate*unmanaged[Fastcall]<int, void> d) => 0;
}
```

---
In reply to: [491271217](https://github.com/dotnet/roslyn/pull/47651#discussion_r491271217) [](ancestors = 491271217)